lib/smack/SmackRep.cpp

```diff
@@ -398,9 +526,9 @@ const Expr* SmackRep::bop(const llvm::BinaryOperator* BO) {
 }
 
 const Expr* SmackRep::bop(unsigned opcode, const llvm::Value* lhs, const llvm::Value* rhs, const llvm::Type* t) {
-  const Expr* e = Expr::fn(bop2fn(opcode),
-                           (isBool(lhs) ? b2i(lhs) : expr(lhs)),
-                           (isBool(rhs) ? b2i(rhs) : expr(rhs)));
+  const Expr* e = Expr::fn(opName(bop2fn(opcode), {(isBool(t)? width : getIntSize(t))}),
```

Comment:
This operation name should depend on the type of `lhs` and `rhs`— which should be the same — and not on the width of pointers, which is irrelevant. See http://llvm.org/docs/LangRef.html#binary-operations .